tests/test_download.py

```diff
@@ -0,0 +1,161 @@
+from enum import Enum, auto
+import unittest
+from unittest.mock import patch, mock_open, call
+
+import pycurl
+
+import download
+
+
+class MockOpts(Enum):
+    URL = auto()
+    WRITEDATA = auto()
+    POSTFIELDS = auto()
+    FOLLOWLOCATION = auto()
+    FAILONERROR = auto()
+
+
+def init_curl_instance(mock_curl):
+    instance = mock_curl.return_value
+    instance.URL = MockOpts.URL
+    instance.FOLLOWLOCATION = MockOpts.FOLLOWLOCATION
+    instance.FAILONERROR = MockOpts.FAILONERROR
+    instance.WRITEDATA = MockOpts.WRITEDATA
+    instance.POSTFIELDS = MockOpts.POSTFIELDS
+    return instance
+
+
+def test_opts(*opts):
+    if not opts:
+        raise Exception("no curl options specified")
+    if len(opts) != 2:
+        raise Exception("expected two args to setopt()")
+    key, val = opts
+    if key == MockOpts.WRITEDATA:
+        val.write(b'foobar')
+
+
+class TestDownload(unittest.TestCase):
+
+    @patch('download.pycurl.Curl')
+    def test_download_get_success_no_dest(self, test_curl):
+        """
+        Test successful GET request when dest is not set.
+        """
+        instance = init_curl_instance(test_curl)
+        instance.setopt.side_effect = test_opts
+        data = download.do_curl("foo")
+        print(test_curl.mock_calls)
```

Comment:
Probably an extra print here.

Reply:
Yep, this was a leftover from debugging the test. I removed it.